### PR TITLE
PMM-4105 Actions UI - part 1.

### PR DESCRIPTION
**Pull request by BupycHuk**
Simple UI to run Explain Action.
image

### CODE REVIEW

File: src/app/qan/profile-details/components/explain/explain.component.ts [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+  ngOnInit() {
+  }
+
+  private startExplainActions(value) {
+    this.startClassic(value);
+    this.startJson(value);
+  }
+
+  private startClassic(value) {
+    this.classicStart$ = this.actionsService.StartMySQLExplainAction({
```
Comment (askomorokhov) [inline src/app/qan/profile-details/components/explain/explain.component.ts]: Is it possible to somehow simplify this nesting?

File: src/app/qan/profile-details/components/explain/explain.component.ts [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    });
+  }
+
+  private startJson(value) {
+    this.jsonStart$ = this.actionsService.StartMySQLExplainJSONAction({
+      service_id: value.service_id,
+      query: value.example,
+      database: value.schema,
+    }).subscribe(item => {
+      const explainResultObservable = interval(1000)
```
Comment (darly-s) [inline src/app/qan/profile-details/components/explain/explain.component.ts]: Is it necessary to use interval here?

File: src/app/qan/profile-details/components/explain/explain.component.ts [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+        .pipe(
+          startWith(0),
+          switchMap(() => this.actionsService.GetAction({ action_id: item.action_id }))
+        )
+        .subscribe(res => {
```
Comment (darly-s) [inline src/app/qan/profile-details/components/explain/explain.component.ts]: Please do not use subscribe in subscribe. You can use switchMap or any else observable pipe operator

File: src/app/qan/profile-details/components/explain/explain.component.ts (hunk 2 above)
Comment (BupycHuk) [reply src/app/qan/profile-details/components/explain/explain.component.ts]: yes, we should pull data from backend each second until we get `done` flag.

File: src/app/qan/profile-details/components/explain/explain.component.ts (hunk 3 above)
Comment (BupycHuk) [reply src/app/qan/profile-details/components/explain/explain.component.ts]: ok

**Comment by darly-s** [comment]
I think need to correct some accordion and table styles, but I'll do it :)

**Comment by BupycHuk** [comment]
Thanks a lot :)

File: Makefile [hunk 4]
```diff
@@ -8,6 +8,6 @@ build:
 	npm run build
 	npm run pack
 deploy:
-	docker exec pmm-server bash -c 'rm -rf /usr/share/percona-qan-app/*'
-	docker cp dist/qan-app pmm-server:/opt/
-	docker exec pmm-server bash -c 'mv /opt/qan-app/* /usr/share/percona-qan-app/'
+	docker exec pmm-server-2-0-0-test1 bash -c 'rm -rf /usr/share/percona-qan-app/*'
```
Comment (BupycHuk) [inline Makefile]: let's revert this

File: src/app/qan/profile-details/components/explain/explain.component.ts [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+      service_id: value.service_id,
+      query: value.example,
+      database: value.schema,
+    }).pipe(switchMap((item) => this.getActionResult(item))).subscribe(
+      res => {
+        if (res.done) {
+          if (!res.error) {
+            this.jsonError = '';
+            this.jsonOutput = JSON.parse(res.output);
+          } else {
```
Comment (BupycHuk) [inline src/app/qan/profile-details/components/explain/explain.component.ts]: let's extract this outside of `if res.done` block

**Pull request merged**